packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -129,10 +129,11 @@ export default function markdown({ config }: AstroPluginOptions): Plugin {
 				// Extract special frontmatter keys
 				let { data: frontmatter, content: markdownContent } = matter(source);
 
-				// Turn HTML comments into JS comments
+				// Turn HTML comments into JS comments while preventing nested `*/` sequences
+				// from ending the JS comment by injecting a zero-width space
 				markdownContent = markdownContent.replace(
 					/<\s*!--([^-->]*)(.*?)-->/gs,
-					(whole) => `{/*${whole}*/}`
+					(whole) => `{/*${whole.replace(/\*\//g, '*\u200b/')}*/}`
```

Comment:
So the problem was with the end `*\` not being properly escaped 🧐

Great catch 🦛💚

Reply:
Thanks! Yeah, the old code wrapped this Markdown code...

```mdx
<!--This text contains which apparen*/tly breaks the compiler in Beta 36.-->
```

...with JS comments like this:

```mdx
This text {/*<!--contains which apparen*/tly breaks the compiler in Beta 36.-->*/}
```

And as you can see in the syntax highlighting above, the contained unescaped `*/` sequence terminated the comment block early, causing the rest to be interpreted as invalid JS code.

My fix escapes the middle `*/` by inserting a zero-width space character into it. I did this to keep the visual result the same but still prevent it from terminating the comment. Unfortunately there seems to be no "proper" way of escaping the end of a block comment, so this is just what I came up with which works. I'm happy with other ways of escaping, too, in case someone wants to change this.